### ISSUE TITLE
IllegalArgumentException in PhoneVerificationActivity

### ISSUE THREAD

**Issue by samtstern** [firebase/FirebaseUI-Android#922]
https://github.com/firebase/FirebaseUI-Android/blob/f4494c4a5fcafc8b32351edcb02ed2834c6a32b4/auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneVerificationActivity.java#L161

Reported by a partner:
```
Fatal Exception: java.lang.IllegalArgumentException: Given String is empty or null
       at com.google.android.gms.common.internal.zzbo.zzcF(Unknown Source)
       at com.google.firebase.auth.PhoneAuthCredential.(Unknown Source)
       at com.google.firebase.auth.PhoneAuthProvider.getCredential(Unknown Source)
       at com.firebase.ui.auth.ui.phone.PhoneVerificationActivity.submitConfirmationCode(PhoneVerificationActivity.java:161)
       at com.firebase.ui.auth.ui.phone.SubmitConfirmationCodeFragment.submitConfirmationCode(SubmitConfirmationCodeFragment.java:175)
       at com.firebase.ui.auth.ui.phone.SubmitConfirmationCodeFragment.access$400(SubmitConfirmationCodeFragment.java:42)
       at com.firebase.ui.auth.ui.phone.SubmitConfirmationCodeFragment$2.onClick(SubmitConfirmationCodeFragment.java:169)
       at android.view.View.performClick(View.java:5706)
       at android.view.View$PerformClick.run(View.java:22834)
       at android.os.Handler.handleCallback(Handler.java:836)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at android.os.Looper.loop(Looper.java:203)
       at android.app.ActivityThread.main(ActivityThread.java:6339)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1084)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:945)
```

**Comment by samtstern** [comment]
The method is called from `SubmitConfirmationCodeFragment`:

```java
private void submitConfirmationCode() {
        mVerifier.submitConfirmationCode(mConfirmationCodeEditText.getUnspacedText().toString());
    }
```

This is called in two places.  One is when the enter button is clicked on the keyboard:

```java
        ImeHelper.setImeOnDoneListener(mConfirmationCodeEditText,
                new ImeHelper.DonePressedListener() {
                    @Override
                    public void onDonePressed() {
                        if (mSubmitConfirmationButton.isEnabled()) {
                            submitConfirmationCode();
                        }
                    }
                });
```

And one is when the submit button in the Activity is clicked:

```java
        mSubmitConfirmationButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                submitConfirmationCode();
            }
        });
```

**Comment by samtstern** [comment]
It does not seem to be possible to get a null confirmation code to get through to this line:

```java
public void submitConfirmationCode(String confirmationCode) {
        showLoadingDialog(getString(R.string.fui_verifying));
        signIn(PhoneAuthProvider.getCredential(mVerificationId, 
}
```

So maybe there is some weird race condition where `mVerificationId` is `null` or empy

**Comment by samtstern** [comment]
This issue has been fixed in version `2.4.0`:
https://github.com/firebase/FirebaseUI-Android/releases/tag/2.4.0

**Comment by arpit7997** [comment]
@samtstern 
java.lang.IllegalArgumentException: 
  at com.google.android.gms.common.internal.Preconditions.checkArgument (Unknown Source)
  at com.google.firebase.auth.PhoneAuthCredential.<init> (Unknown Source)
  at com.google.firebase.auth.PhoneAuthProvider.getCredential (Unknown Source)
  at com.online.suvidhasupermarket.login.EnterOTPActivity.verifyCode (Unknown Source)
  at com.online.suvidhasupermarket.login.EnterOTPActivity.a (Unknown Source)
  at com.online.suvidhasupermarket.login.EnterOTPActivity$1.onClick (Unknown Source)
  at android.view.View.performClick (View.java:5231)
  at android.view.View$PerformClick.run (View.java:21240)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:179)
  at android.app.ActivityThread.main (ActivityThread.java:5730)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:791)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:681)

I'm getting this error on Oppo CPH1701 (CPH1701), Android 6.0,Samsung Galaxy C7 Pro (c7prolte), Android 8.0,Samsung Galaxy C9 Pro (c9lte), Android 8.0

Please help me resolve this issue.
